instance/instance.go

```diff
@@ -0,0 +1,139 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+// Package instance implements VM instance software configs.
+package instance
+
+import (
+	"bytes"
+	"encoding/json"
+
+	agentendpointpb "github.com/GoogleCloudPlatform/osconfig/_internal/gapi-cloud-osconfig-go/google.golang.org/genproto/googleapis/cloud/osconfig/agentendpoint/v1alpha1"
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/golang/protobuf/proto"
+)
+
+// LocalConfig represents the structure of the config to the JSON parser.
+//
+// The types of members of the struct are wrappers for protobufs and delegate
+// the parsing to jsonpb lib via their UnmarshalJSON implementations.
+type LocalConfig struct {
+	Packages            []Package
+	PackageRepositories []PackageRepository
+	SoftwareRecipes     []SoftwareRecipe
+}
+
+type Package struct {
+	i agentendpointpb.Package
+}
+
+func (r *Package) UnmarshalJSON(b []byte) error {
+	rd := bytes.NewReader(b)
+	return jsonpb.Unmarshal(rd, &r.i)
+}
+
+type PackageRepository struct {
+	i agentendpointpb.PackageRepository
+}
+
+func (r *PackageRepository) UnmarshalJSON(b []byte) error {
+	rd := bytes.NewReader(b)
+	return jsonpb.Unmarshal(rd, &r.i)
+}
+
+type SoftwareRecipe struct {
+	r agentendpointpb.SoftwareRecipe
+}
+
+func (r *SoftwareRecipe) UnmarshalJSON(b []byte) error {
+	rd := bytes.NewReader(b)
+	return jsonpb.Unmarshal(rd, &r.r)
+}
+
+func ParseLocalConfig(a []byte) (rv LocalConfig, e error) {
+	e = json.Unmarshal(a, &rv)
```

Comment:
This should work in one line:
return rv, json.Unmarshal(a, &rv)

Reply:
Done.